engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/AbstractConstraintValidatorContext.java

```diff
@@ -273,16 +273,16 @@ private CrossParameterConstraintViolationBuilderImpl(List<String> methodParamete
 
 		@Override
 		public NodeBuilderDefinedContext addParameterNode(int index) {
-			if ( propertyPath.getLeafNode().getKind() != ElementKind.CROSS_PARAMETER ) {
-				throw LOG.getParameterNodeAddedForNonCrossParameterConstraintException( propertyPath );
-			}
```

Comment:
As mentioned in my earlier comment, I think I would add a `Contracts.assertTrue()` condition to the constructor.